devito/types/dense.py

```diff
@@ -1104,7 +1107,8 @@ def __halo_setup__(self, **kwargs):
                 halo = (left_points, right_points)
             else:
                 raise TypeError("`space_order` must be int or 3-tuple of ints")
-            return tuple(halo if i.is_Space else (0, 0) for i in self.dimensions)
+            halo = tuple(halo if i.is_Space else (0, 0) for i in self.dimensions)
+            return DimensionTuple(*halo, getters=self.dimensions)
```

Comment:
you can probably factor 

```
return DimensionTuple(...)
```

out of the `if-else`. This way, you'll also catch the case at line 1100, which might still return a plain tuple otherwise